### PR TITLE
Jump Search in Python

### PR DESCRIPTION
**Pull request by himanshu007-creator**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

This pr solves issue #1724 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1724

### CODE REVIEW

**Review by atarax665 — COMMENTED** [review]
Please follow our standard [Python Code Template](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/Code_Template.md).

**Comment by himanshu007-creator** [comment]
> Please follow our standard [Python Code Template](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/Code_Template.md).

sir i cant figure out whats wrong

**Comment by HarshCasper** [comment]
> > Please follow our standard [Python Code Template](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/Code_Template.md).
> 
> sir i cant figure out whats wrong

Use a tool called `PyLint` to see all the formatting and stylistic errors in the Code. Kindly resolve all the generated warnings. Also have a look at the merged PRs to understand how to format the Code properly. 

Thanks! Do let me know if you need more clarity.

File: Python/search/jump_search.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    else:
+        arr_ls = arr[chunk-interval:chunk]
+        ind = [i for i,d in enumerate(arr_ls) if d==search]
+        return chunk-interval+ind[0]
+#driver code
+arr = list(map(int,input().split()))
+#get the item to be searched
+val=int(input())
+res = jump_search(arr, val)
+print(res)
```
Comment (HarshCasper) [inline Python/search/jump_search.py]: Enclose in a `if __name__ == '__main__'` block

File: Python/search/jump_search.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# The basic idea is to check fewer elements by jumping ahead by fixed steps or skipping some elements in place of searching all elements.
+# Time Complexity : O(√n)
+# Auxiliary Space : O(1)
+
+import math
+#main function
+def jump_search(arr,search):
```
Comment (HarshCasper) [inline Python/search/jump_search.py]: Add a Function Docstring

File: Python/search/jump_search.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        arr_ls = arr[chunk-interval:chunk]
+        ind = [i for i,d in enumerate(arr_ls) if d==search]
+        return chunk-interval+ind[0]
+#driver code
+arr = list(map(int,input().split()))
+#get the item to be searched
+val=int(input())
+res = jump_search(arr, val)
+print(res) 
+# NOTE:-Works only sorted arrays.
```
Comment (HarshCasper) [inline Python/search/jump_search.py]: Add this on the top of the Code as a Comment

File: Python/search/jump_search.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# The basic idea is to check fewer elements by jumping ahead by fixed steps or skipping some elements in place of searching all elements.
+# Time Complexity : O(√n)
+# Auxiliary Space : O(1)
```
Comment (HarshCasper) [inline Python/search/jump_search.py]: Keep the Time and Space Complexity at the bottom along with a sample I/O

File: Python/search/jump_search.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    if flag==0:
+        c=i
+        for j in arr[i:]:
+            if j==search:
+                return c
+            c+=1
+    else:
+        arr_ls = arr[chunk-interval:chunk]
+        ind = [i for i,d in enumerate(arr_ls) if d==search]
+        return chunk-interval+ind[0]
```
Comment (HarshCasper) [inline Python/search/jump_search.py]: Format your Code properly.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/search/jump_search.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+if __name__ == '__main__':
+    ary = list(map(int, input().split()))
+    val= int(input())
+    res = jump_search(ary, val)
+    print(res)    
+# Time Complexity : O(√n)
+# Auxiliary Space : O(1)
+# SAMPLE INPUT line 1   1 2 3 4 5 6 7 8 9 10
+# SAMPLE INPUT line 2   5
+# SAMPLE OUTPUT     4
```
Comment (HarshCasper) [inline Python/search/jump_search.py]: Keep it in a Multi-Line String

File: Python/search/jump_search.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# JUMP-SEARCH
+# NOTE:-Works only sorted arrays.
+import math
+# main function
+def jump_search(arr, search):
```
Comment (HarshCasper) [inline Python/search/jump_search.py:15]: Add a function docstring

File: Python/search/jump_search.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
+# JUMP-SEARCH
+# NOTE:-Works only sorted arrays.
```
Comment (HarshCasper) [inline Python/search/jump_search.py]: Add a better pitch to the code

File: Python/search/jump_search.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
+# JUMP-SEARCH
+# NOTE:-Works only sorted arrays.
+import math
+# main function
+def jump_search(arr, search):
+    
+
```
Comment (HarshCasper) [inline Python/search/jump_search.py]: Remove these extra lines of code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by himanshu007-creator** [comment]
Done with changes. Please add <code>DWOC</code> and <code>Level</code> tags accordingly.

File: Python/search/jump_search.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
+"""
+NOTE:-Works only sorted arrays.
```
Comment (atarax665) [inline Python/search/jump_search.py:4]: Add a small description Jump search and an insight on how it works.

File: Python/search/jump_search.py [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+            break
+    l_index = [e for e, i in enumerate(arr[low:low+interval]) if i == search]
+    if l_index[0]:
+        return low+l_index[0]
+    return "Not Found"
+
+
+# driver code
+if __name__ == "__main__":
+    ary = list(map(int, input().split()))
```
Comment (atarax665) [inline Python/search/jump_search.py]: Use appropriate prompts for taking input and returning output.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/search/jump_search.py [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    for i in range(0, len(arr), interval):
+        if arr[i] < search:
+            low = i
+        elif arr[i] == search:
+            return i
+        else:
+            break
+    l_index = [e for e, i in enumerate(arr[low:low+interval]) if i == search]
+    if l_index[0]:
+        return low+l_index[0]
```
Comment (atarax665) [inline Python/search/jump_search.py]:
```suggestion
        return low + l_index[0]

```

File: Python/search/jump_search.py (hunk 11 above)
Comment (himanshu007-creator) [reply Python/search/jump_search.py]: done with the changes

**Review by atarax665 — COMMENTED** [review]
Update Readme.md

File: Python/search/jump_search.py (hunk 12 above)
Comment (himanshu007-creator) [reply Python/search/jump_search.py]: done

**Comment by himanshu007-creator** [comment]
> Update Readme.md

Done👍

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**